lib/public/models/DataExportModel.js

```diff
@@ -0,0 +1,207 @@
+     *
+     * @return {DataExportConfiguration} configuration
+     */
+    get dataExportConfiguration() {
+        return this._dataExportConfiguration;
+    }
+
+    /**
+     * Get total number of items that model knows they exist
+     *
+     * @return {number} totalExistingItemsCount
+     */
+    get totalExistingItemsCount() {
+        return this._totalExistingItemsCount;
+    }
+
+    /**
+     * Set total number of items that model should they exist
+     * @param {number} totalExistingItemsCount count
+     */
+    setTotalExistingItemsCount(totalExistingItemsCount) {
+        this._totalExistingItemsCount = totalExistingItemsCount;
+    }
+
+    /**
+     * State whether exporting is enabled
+     * @return {boolean} if true disabled, otherwise enabled
+     */
+    get disabled() {
+        return this._disabled;
+    }
+
+    /**
+     * Set whether exporting is enabled
+     * @param {boolean} disabled if true disabled, otherwise enabled
+     */
+    setDisabled(disabled) {
+        this._disabled = disabled;
+    }
+
+    /**
+     * If the model has no data to be exported, it calls for it via action function provided in constructor
+     * @return {Promise<void>|null} promise if data were not asked for, null otherwise
+     */
+    callForData() {
+        if (this._onDataNotAskedAction) {
+            return this.items.match({
+                NotAsked: () => this._onDataNotAskedAction(),
+                Other: () => null,
+            });
+        } else {
+            throw new Error('onDataNotAskedAction was not provided');
+        }
+    }
+
+    /**
+     * Return the current items remote data
+     *
+     * @return {RemoteData<T[]>} the items
+     */
+    get items() {
+        return this._items$.getCurrent();
+    }
+
+    /**
+     * Observable notified when the export configuration visually changes
+     *
+     * @return {Observable} the visual change observable
+     */
+    get visualChange$() {
+        return this._visualChange$;
+    }
+
+    /**
+     * Get export type selected by the user
+     *
+     * @return {string} export type
+     */
+    get selectedExportType() {
+        return this._selectedExportType;
+    }
+
+    /**
+     * Set export type
+     *
+     * @param {string} exportType export type
+     * @return {void}
+     */
+    setSelectedExportType(exportType) {
+        this._selectedExportType = exportType;
+        this.notify();
+        this._visualChange$.notify();
+    }
+
+    /**
+     * Get selected fields
+     *
+     * @return {string[]} selected fields
+     */
+    get selectedFields() {
+        return this._selectedFields;
+    }
+
+    /**
+     * Update selected fields from HTML options list
+     *
+     * @param {HTMLCollection|Array} selectedOptions options collection
+     * @return {void}
+     */
+    setSelectedFields(selectedOptions) {
+        this._selectedFields = [];
+        [...selectedOptions].forEach(({ value }) => this._selectedFields.push(value));
```

Comment:
Would it not be better to use a Set to avoid potential duplicates?
Also, you can use the `Array.from` method (standalone or with a set) instead of the `forEach` for improved readability